WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -160,15 +160,15 @@ extension PointOfSaleAggregateModel {
 
     func connectCardReader() {
         analytics.track(.pointOfSaleCardReaderConnectionTapped)
-        Task { @MainActor in
-            _ = try await cardPresentPaymentService.connectReader(using: .bluetooth)
+        Task { @MainActor [weak self] in
+            _ = try await self?.cardPresentPaymentService.connectReader(using: .bluetooth)
```

Comment:
Does this really form part of a retain cycle? We don't retain the `Task`, so I'm not sure why this is needed...

Reply:
Tasks are not released until the asynchronous work inside it finishes.

As I tested, `attemptConnection` never finishes when onboarding is opened and closed. `Task` is not released until the `connectReader` finishes and it doesn't finish. Since `Task` retains `self` through accessing `cardPresentPaymentService`, `self` (`PointOfSaleAggregateModel`) remains retained. 

https://github.com/user-attachments/assets/ccc80bb9-4b1d-4ef8-98c7-953b057bf580